crates/era-downloader/src/client.rs

```diff
@@ -0,0 +1,102 @@
+use futures::StreamExt;
+use reqwest::{Client, IntoUrl, Url};
+use std::{path::Path, str::FromStr};
+use tokio::{fs::File, sync::mpsc};
+
+#[derive(Debug)]
+pub struct EraClient {
+    client: Client,
```

Comment:
I'd like to structure this slightly differently,

by introducing a standalone `EraClient(Client)` we can get a nice to use reqwest wrapper type that provides easy access to era download functionality, which i think could also be nice for testing